static/js/components/forms/validation.ts

```diff
@@ -0,0 +1,42 @@
+import * as yup from "yup"
+import { SchemaOf, setLocale } from "yup"
+
+import { ConfigItem } from "../../types/websites"
+
+// This is added to properly handle file fields, which can have a "null" value
+setLocale({
+  mixed: {
+    notType: "${path} is a required field."
+  }
+})
+
+const defaultYupField = yup.string()
+const widgetToYupMap = {
+  string:   yup.string(),
+  text:     yup.string(),
+  markdown: yup.string(),
+  file:     yup.object().shape({
+    name: yup.string()
+  })
+}
+
+export const getContentSchema = (configItem: ConfigItem): SchemaOf<any> => {
+  const yupObjectShape = {}
+  configItem.fields.forEach(field => {
+    const yupField = widgetToYupMap[field.widget] || defaultYupField
+    yupObjectShape[field.name] = yupField.label(field.label)
+    if (field.required) {
+      switch (field.widget) {
+      case "file":
+        yupObjectShape[field.name] = yupObjectShape[field.name].fields.name
```

Comment:
I'm seeing an error when adding a new file to a resource or editing an existing resource:

![Screenshot from 2021-03-26 16-34-30](https://user-images.githubusercontent.com/863262/112689966-613fd300-8e51-11eb-9649-c7f69e45b782.png)

Reply:
Are you sure you pulled down the latest and the bundle rebuilt? I'm not seeing that...

Reply:
It was an outdated schema. Running the management command fixed it